src/poetry/core/constraints/version/empty_constraint.py

```diff
@@ -22,6 +22,9 @@ def is_any(self) -> bool:
     def is_simple(self) -> bool:
         return True
 
+    def has_upper_bound(self) -> bool:
+        return True
```

Comment:
Sanity question, does an empty constraint have a max?

Reply:
Good question. My thinking was that the opposite of an empty constraint, which is `*`, has no upper bound and the two extremes often behave the other way around. Another explanation might be that if there is no version that can satisfy the constraint, this is like an upper bound of `0` (not included).

Further, it should practically irrelevant - at least for now.

Reply:
That sort of makes sense to me. 

Nitpick, maybe add a comment in there so someone doesn't change it later?